src/components/EventDetails/EventDetails.js

```diff
@@ -18,862 +18,1089 @@
-      const toMyself = isReceived && addressesEqual(from, to);
-      let transactionNote = note;
-      if (txNotes && txNotes.length > 0) {
-        const txNote = txNotes.find(txn => txn.txHash === eventData.hash);
-        if (txNote) {
-          transactionNote = txNote.text;
-        }
-      }
-      const hasNote = transactionNote && transactionNote !== '';
-      const isPending = status === TX_PENDING_STATUS;
-      const assetsData = getAssetsAsList(assets);
-      const { decimals = 18 } = getAssetData(assetsData, supportedAssets, asset);
-      const value = formatUnits(txInfo.value, decimals);
-      const recipientContact = this.findMatchingContactOrAccount(to);
-      const senderContact = this.findMatchingContactOrAccount(from);
-      const relatedAddress = isReceived ? from : to;
-      const relatedUser = isReceived ? senderContact : recipientContact;
-      let relatedUserTitle = relatedUser.username
-        // $FlowFixMe
-        || getAccountName(relatedUser.type)
-        || ensRegistry[relatedAddress]
-        || relatedAddress;
-      if (addressesEqual(to, from)) {
-        relatedUserTitle = 'My account';
-      }
-      const relatedUserProfileImage = relatedUser.profileImage || null;
-      // $FlowFixMe
-      const showProfileImage = !relatedUser.type;
+    const formattedValue = formatAmount(value);
 
-      const fee = gasUsed && gasPrice ? Math.round(gasUsed * gasPrice) : 0;
-      const freeTx = isPPNTransaction || tag === SET_SMART_WALLET_ACCOUNT_ENS;
+    const directionIcon = isReceived ? 'received' : 'sent';
+    let directionSymbol = isReceived ? '+' : '-';
 
-      const showFeeBlock = (toMyself || !isReceived) && !isPending && (freeTx || !!fee);
-      let showNote = true;
-      const listSettledAssets = (tag === PAYMENT_NETWORK_TX_SETTLEMENT && !isEmpty(extra));
+    if (formattedValue === '0') {
+      directionSymbol = '';
+    }
 
-      if (tag === PAYMENT_NETWORK_TX_SETTLEMENT
-        || tag === PAYMENT_NETWORK_ACCOUNT_TOPUP
-        || tag === PAYMENT_NETWORK_ACCOUNT_WITHDRAWAL
-      ) {
-        showNote = false;
-      }
+    const isPending = isPendingTransaction(event);
+
+    let eventData: ?EventData = null;
+
+    switch (event.tag) {
+      case PAYMENT_NETWORK_ACCOUNT_DEPLOYMENT:
+        return {
+          name: 'Smart Wallet',
+          itemImageSource: smartWalletIcon,
+          actionTitle: 'Activated',
+          actionSubtitle: this.getFeeLabel(),
+          buttons: [
+            {
+              title: 'Activate Pillar Network',
+              onPress: this.topUpPillarNetwork,
+              secondary: true,
+            },
+            {
+              title: 'Refer friends',
+              onPress: this.referFriends,
+              squarePrimary: true,
+            },
+          ],
+        };
+      case PAYMENT_NETWORK_ACCOUNT_TOPUP:
+        if (activeBlockchainNetwork === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK) {
```

Comment:
Hm, I feel like this is not fair to the user to show this info only if PPN is active. We could change active network if user pick any of those actions (just like we do with other events).

Now, options (buttons) availability is another question. Send availability should be based on PPN's balance and "Top up more", on PLR balance in Smart Wallet.

Reply:
In Activity Feed it's like this: if we're on PillarNetwork, we display that it's a top up with Pillar Network. If we're on SW, we display it as a transaction from SW to PN. So I made it similar now: if we're on PN, I show this modal, if we're on SW, I show the usual transaction modal.